### ISSUE TITLE
ShardingSphere-JDBC readwrite-splitting config bug

### ISSUE THREAD

**Issue by wzkris** [apache/shardingsphere#26399]
### Which version of ShardingSphere did you use?
5.3.2
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-JDBC
### Expected behavior
i configure it exactly according to the official configuration like this.
![image](https://github.com/apache/shardingsphere/assets/90759163/139a1453-90b5-4511-8d7e-5575f97b530d)
### Actual behavior
![image](https://github.com/apache/shardingsphere/assets/90759163/5d77924c-3eb0-470f-a56e-79eb0ecb5ef8)
### Reason analyze (If you can)
I debug the source code found that may occured in MetaDataContextFactory class.
![image](https://github.com/apache/shardingsphere/assets/90759163/d4d63d94-70e3-4283-be40-6591284952ef)
In the code line 45, you can see my readwrite-splitting config is totally correct, but in line 46, the variable `effectiveDatabaseConfigs `changed.
![image](https://github.com/apache/shardingsphere/assets/90759163/d9863c53-1e65-41be-9ac9-5c27ac9606ef)
The method `createEffectiveDatabaseConfigurations` should read my config, Why it didn't read into the parameter configuration？

**Comment by jiangML** [comment]
@wzkris Hi, if cluster mode is configured, the governance center will prevail.

**Comment by wzkris** [comment]
> @wzkris Hi, if cluster mode is configured, will use the configuration of the governance center

@jiangML Yes, I configured the cluster mode with zookeeper, does this mean that the yml config file only needs to configure the cluster, and the rest of the configuration is configured in zookeeper

**Comment by zhaojinchao95** [comment]
> > @wzkris Hi, if cluster mode is configured, will use the configuration of the governance center
> 
> @jiangML Yes, I configured the cluster mode with zookeeper, does this mean that the yml config file only needs to configure the cluster, and the rest of the configuration is configured in zookeeper

Yes, you are right.

**Comment by wzkris** [comment]
@zhaojinchao95 I use druid pool , but why HikariPool still running.
![image](https://github.com/apache/shardingsphere/assets/90759163/7cadb541-fd07-42cd-9e6a-f3cd9fef9b8a)

**Comment by Duria-boy** [comment]
> @zhaojinchao95 I use druid pool , but why HikariPool still running. ![image](https://user-images.githubusercontent.com/90759163/246707475-1b26f929-928a-497d-8274-f05482d727dc.png)
I think this HikariPool is metadata datasource for shardingshpere , and not related to the your druid database.
